### PR TITLE
Improve tests

### PR DESCRIPTION
**Pull request by Milly**
- All client tests have been rewritten. related #19
- Change throws to rejects in test names.
- Add missing tests. 100% coverage.
- Fix example.

## Summary by CodeRabbit

- **New Features**
  - Improved error handling in session reservations for increased reliability.
- **Bug Fixes**
  - Updated test descriptions and added new test cases for better error management.
- **Refactor**
  - Renamed `Session` to `SessionForClient` for clarity in client contexts.
- **Tests**
  - Enhanced testing frameworks with new stubs and spies for precise client behavior verification.
- **Documentation**
  - Adjusted export statements for clearer module interactions.

### CODE REVIEW

File: client.ts [hunk 1]
```diff
@@ -11,7 +11,7 @@ import {
 
 const msgidThreshold = 2 ** 32;
 
-type Session = {
+export type SessionForClient = {
```
Comment (lambdalisue) [inline client.ts]: `SessionForClient` in `client.ts` is a bit redundant. It should not be renamed.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -39,7 +39,7 @@ jobs:
           deno task test
         timeout-minutes: 5
       - run: |
-          deno coverage --unstable .coverage --lcov > coverage.lcov
+          deno coverage .coverage --lcov > coverage.lcov
```
Comment (lambdalisue) [inline .github/workflows/test.yml]: Please use `deno task coverage --lcov > coverage.lcov` instead

**Review by lambdalisue — CHANGES_REQUESTED** [review]

File: client.ts (hunk 1 above)
Comment (Milly) [reply client.ts]: There is a name conflict in mod.ts, what should I do?

File: session.ts [hunk 3]
```diff
@@ -92,7 +92,11 @@ export class Session {
       return Promise.reject(new Error("Session is not running"));
     }
     const { reservator } = this.#running;
-    return reservator.reserve(msgid);
+    try {
+      return reservator.reserve(msgid);
+    } catch (err) {
+      return Promise.reject(err);
+    }
```
[the remark this answers is not in the document]
Comment (Milly) [reply session.ts]: `reservator.reserve()` throws an error instead to returns rejected promise.

File: client.ts (hunk 1 above)
Comment (lambdalisue) [reply client.ts]: Do we need to export `Session` in `client.ts` as public? I thought it was required for internal unit tests.

Comment (Milly) [reply client.ts]: It is argument type of public class.
The types of public arguments of public methods in public classes should be public.

Comment (lambdalisue) [reply client.ts]: > It is argument type of public class.
The types of public arguments of public methods in public classes should be public.

I'm sorry but it doesn't make sense to me. I couldn't imagine developers who use this library need the `SessionForClient` while we expose a real `Session` that is compatible with `SessionForClient`.

Could you explain the reason why you would like to expose the `SessionForClient` a bit more?

Comment (Milly) [reply client.ts]: got it

**Review by lambdalisue — APPROVED** [review]
LGTM

**Comment by lambdalisue** [comment]
@Milly Could you rebase to tidy commits? The PR includes test improvements and bug fix so I'd like to add merge commit rather than squash merge.

**Comment by Milly** [comment]
I squashed the commits into units.

**Pull request merged**